src/modules/projectSearchContext.js

```diff
@@ -11,13 +11,112 @@ class ProjectSearchContext {
         this.projectId = projectId;
     }
 
+    static #tryObjectParse = (buffer, callback) => {
+        let start = 0;
+        let depth = 1;
+        let inEscape = false;
+        let inQuote = false;
+        for (let i = 1; i < buffer.length; i++) {
```

Comment:
Maybe check buffer.length > 2 to prevent IndexOfOutBounds errors?

Reply:
Good idea, added